app/api/notify.py

```
@@ -25,12 +25,16 @@
 import datetime
 from datetime import timedelta
 import smtplib
-
+from app.modules.phone_util import create_sms,Push_notification
+from app.modules.template_util import assign_letter_heads, construct_template, attach_letter_head,generate_full_template_from_string_payload
+from app.modules.sendmail_util import create_sender_list
 bp = Blueprint('notify', __name__, url_prefix='/notify')
         
+
+#dispatch was made to send particular message on slack but those message can be sended on email too
 @bp.route('/dispatch', methods=["POST"])
 #@token.authentication
-def dispatch():
+def construct_dispatch_notification():
     if not request.json:
         abort(500)
     MSG_KEY = request.json.get("message_key", None)

```

create an folder called 'model' and put there a file name notification_msg.py

create function 

get_notification_function_by_key()

    message_detail = mongo.db.notification_msg.find_one({"message_key": MSG_KEY})

and move this code there.

also if messages_Details doesn't exist it should throw an exception.

if message_detail and message_detail['message_type'] is not None
so this if condition is not needed

---

in util. create a function

contruct_payload_from_request.

message = message_detail['message']
            missing_payload = []
            # looping over all the needs check if my message type in that key and if found
            for key in message_needs:
                if message_detail['message_type'] == key:
                    need_found_in_payload = False
                    # LOOP OVER THE KEYS inside the need FOR REQUEST
                    for data in message_needs[key]:
                        need_found_in_payload = False
                        if data in request.json:
                            need_found_in_payload = True
                        # REQUIREMNT DOES NOT SATISFIED RETURN INVALID REQUEST
                        else:
                            missing_payload.append(data)
                            # return jsonify(data + " is missing from request"), 400
                    # IF FOUND PROCESS THE REQUEST.JSON DATA

move ths code there.

it should throw exception again, if request is not satisfied

---

done